packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -1191,6 +1191,23 @@ enum ResizeImagePolicy {
 /// Instructs Flutter to decode the image at the specified dimensions
 /// instead of at its native size.
 ///
+/// The [width] and [height] parameters refer to logical pixels.
+///
+/// Logical pixels are roughly the same visual size across devices.
+/// Physical pixels are the size of the actual hardware pixels on the device.
+/// The number of physical pixels per logical pixel is described by the `devicePixelRatio`.
+///
+/// For example, if you want to resize the image to half of the screen's width, you can use:
```

Comment:
From the [style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-the-passive-voice-recommend-do-not-require-never-say-things-are-simple):

> Never use "you" or "we".

<br>

I think it'd be nice to move the snippet to the bottom of the documentation and to wrap the section using [dartdoc](https://pub.dev/documentation/snippets/latest/) syntax:

````dart
/// cached size provided here.
/// 
/// {@tool snippet}
/// This example shows how to size the image to half of the screen's width.
/// 
/// ```dart
/// // [sample code]
/// ```
/// {@end-tool}
````